airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/SpillToDiskTask.kt

```diff
@@ -59,60 +58,52 @@ class DefaultSpillToDiskTask(
     }
 
     override suspend fun execute() {
-        do {
```

Comment:
doublechecking: we don't need to loop here anymore b/c we're reeunqueueing the spill task somewhere else? (pretty sure I saw that in this PR, but I've already forgotten where 😅 )

Reply:
Yes, that was another change I should have left for a future PR. Sorry. (I owe you a drink.)

But yeah: `handleNewSpilledFile` now re-dispatches SpillToDiskTask as long as end-of-stream hasn't been hit.

The practical reason is that I want each run to be guarded by a check as to whether the stream has been killed.

High-level tho, the contract here should be that tasks shouldn't do flow control. They should run once and yield one result. Let the TaskLauncher decide who should rerun when.